File: 189-rotate-array/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        k%=nums.length;
        int[] b= new int[nums.length];
        for(int i=0;i<nums.length;i++){
            b[(i+k)%nums.length]=nums[i];
        }
        for(int i=0;i<nums.length;i++){
            nums[i]=b[i];
        }
    }
}

File: 189-rotate-array/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        def helper(l,r,arr):
            while l<r:
                arr[l],arr[r]=arr[r],arr[l]
                l+=1
                r-=1
        k=k%len(nums)
        helper(0,len(nums)-1,nums)
        helper(0,k-1,nums)
        helper(k,len(nums)-1,nums)
        
        
        